### ISSUE TITLE
Logs are not printed unless `DEBUG=True` is set

### ISSUE THREAD

**Issue by neubig** [All-Hands-AI/openhands-resolver#235]
Currently logs are not printed unless DEBUG=True is set.

According to @rbren , we may have to explicitly subscribe to the event stream and decide what to print, like:

```python
def on_event(evt):
  print(evt)

event_stream.subscribe(on_event, EventStreamSubscriber.MAIN)
```

Here's a reference: https://github.com/All-Hands-AI/OpenHands/blob/b4afd9f17061cd2fa90563bb46ee5df74667118a/openhands/core/main.py#L190

In terms of next steps, we should probably:
1. Inspect the code of the All-Hands-AI/OpenHands repo in the appropriate place.
2. Add event stream printing code in the `resolve_issues.py` of this repo.
3. Add tests to make sure that it works.

**Comment by malhotra5** [comment]
FYI this may not be resolved in openhands-resolver alone. There are two changes

# Changes required

## Fix 1
In `All-Hands-AI/OpenHands` in `openhands/core/main/run_controller.py` change line 190

```
- event_stream.subscribe(EventStreamSubscriber.MAIN, on_event)
+ event_stream.subscribe(EventStreamSubscriber.MAIN, on_event, append=True)
```

## Fix 2
In `All-Hands-AI/OpenHands` in `openhands/events/stream.py` around line 157

```
- callback = stack[-1]
+ for callback in stack:
+     logger.info("added tasks")
+     tasks.append(asyncio.create_task(callback(event)))

```
 ## Fix 3
 In `All-Hands-AI/openhands-resolver`
 
```
   await runtime.connect()
+  async def on_event(evt):
+     logger.info(evt)
+     runtime.event_stream.subscribe(EventStreamSubscriber.MAIN, on_event, append=True)
```

# Explanation

1. Openhands resolver needs to subscribe to the event stream (Fix #3)
2. Openhands resolver now subscribes to `MAIN`. After this`run_controller` is invoked, which calls openhands proper and subscribes to `MAIN`. This causes an error because multiple callbacks are being added (fix #1)
3. Even though openhands-resolver subscribed to MAIN, logs are being called. This is openhands only invokes the last callback that subscribed to `MAIN`. The fix was to create a task for all callbacks (fix #2)

**Comment by neubig** [comment]
Thanks @malhotra5 ! A PR is welcome!

**Comment by malhotra5** [comment]
@d-walsh I believe that isn't an issue and functions as expected :)

**Comment by neubig** [comment]
Fixed by #237
